File: accounts/accounts-domain/src/main/java/com/hrrm/famoney/accounts/AccountsDomainEntity.java
package com.hrrm.famoney.accounts;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.ToString;
import org.jetbrains.annotations.NotNull;
import org.springframework.data.domain.Persistable;

import javax.persistence.*;

@MappedSuperclass
@EqualsAndHashCode
@Getter
@ToString
public abstract class AccountsDomainEntity implements Persistable<Integer> {

    public static final String SCHEMA_NAME = "accounts";

    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @NotNull
    private Integer id;

    @Override
    public boolean isNew() {
        return id == null;
    }

}

File: accounts/accounts-domain/src/main/kotlin/com/hrrm/famoney/accounts/AccountsDomainEntity.kt
package com.hrrm.famoney.accounts

import lombok.EqualsAndHashCode
import lombok.Getter
import lombok.ToString
import org.springframework.data.domain.Persistable
import javax.persistence.*
import kotlin.properties.Delegates

@MappedSuperclass
abstract class AccountsDomainEntity {
    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int? = null

    companion object {
        const val SCHEMA_NAME = "accounts"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as AccountsDomainEntity

        if (id == null || other.id == null || id != other.id) return false

        return true
    }

    override fun hashCode(): Int {
        return id ?: 0
    }
}